protocol-units/execution/maptos/opt-executor/src/executor.rs

```diff
@@ -282,14 +286,19 @@ impl Executor {
 			)?;
 		} 
 
-		{
+		let proof = {
 			let reader = self.db.read().await.reader.clone();
-			let proof = reader.get_state_proof(
+			reader.get_state_proof(
 				state_compute.version(),
-			)?;
-		}
+			)?
+		};
 
-		Ok(())
+		let commitment = Commitment::digest_state_proof(&proof);
+		Ok(BlockCommitment {
+			block_id: Id(block_id.to_vec()),
+			commitment,
+			height: latest_version,
```

Comment:
This is probably bonkers. Should the current height be available in the executor state? I assume it is stateful and can only execute blocks in a single chain.

Reply:
Yes, there's only a single chain that should be considered inside `execute_block`, no forks that you have to consider. All of that will be handled outside by reverting blocks and the like.

I think `latest_version` should be valid. One test to write is to check that it is increasing with no gaps, i.e.,

$$
f(k) = k : k \in N
$$

$$
f(n) = f (n - 1) + 1 
$$